### PR TITLE
Refactor subnets subscriptions.

### PR DESCRIPTION
**Pull request by nalepae**
**What type of PR is this?**
Other

**What does this PR do? Why is it needed?**
This pull requests refactors subnets subscriptions.

It removes `subscribeDynamicWithSubnets` and `subscribeStaticWithSyncSubnets` to only use `subscribeDynamicWithSubnets ` and `subscribeStaticWithSubnets`.

It also paves the way of future data columns (for peerDAS) subscriptions without the need to add specifics functions.

**Acknowledgements**
- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have made an appropriate entry to [CHANGELOG.md](https://github.com/prysmaticlabs/prysm/blob/develop/CHANGELOG.md).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

### CODE REVIEW

File: beacon-chain/sync/subscriber.go [hunk 1]
```diff
@@ -87,18 +107,23 @@ func (s *Service) registerSubscribers(epoch primitives.Epoch, digest [4]byte) {
 	)
 	if flags.Get().SubscribeToAllSubnets {
 		s.subscribeStaticWithSubnets(
+			"Attestation",
```
Comment (nisdas) [inline beacon-chain/sync/subscriber.go:110]: Can we provide the actual topic or object type here ? Its better than using hardcoded strings here

File: beacon-chain/sync/subscriber.go [hunk 2]
```diff
@@ -109,19 +134,26 @@ func (s *Service) registerSubscribers(epoch primitives.Epoch, digest [4]byte) {
 			s.syncContributionAndProofSubscriber,
 			digest,
 		)
+
 		if flags.Get().SubscribeToAllSubnets {
-			s.subscribeStaticWithSyncSubnets(
+			s.subscribeStaticWithSubnets(
+				"Sync committee",
```
Comment (nisdas) [inline beacon-chain/sync/subscriber.go:140]: We already add in the `SyncCommitteeSubnetTopicFormat` topic, so do we need to add this again ?

File: beacon-chain/sync/subscriber.go (hunk 1 above)
Comment (nalepae) [reply beacon-chain/sync/subscriber.go:110]: This `description` string is only used in:
```go
log.WithField("digest", digest).Infof("%s subnets with are no longer valid, unsubscribing from all of them.", description)
```

Before, we had the log hardcoded version:

```go
log.Warnf("Attestation subnets with digest %#x are no longer valid, unsubscribing from all of them.", digest)
```

So, it was already hard-coded before, but not at the same location.
I can for sure use the actual topic instead, but the log will be less human readable.

File: beacon-chain/sync/subscriber.go (hunk 2 above)
Comment (nalepae) [reply beacon-chain/sync/subscriber.go:140]: See other comment.

File: beacon-chain/sync/subscriber.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/sync/subscriber.go:110]: Could you possible use a helper function with a switch of all topics to derive this (ex: If the topic is the attestation subnet, then the description will always be `Attestation` . And so on. This way we don't need to add in an additional argument

File: beacon-chain/sync/subscriber.go [hunk 3]
```diff
@@ -620,38 +453,20 @@ func (s *Service) subscribeToSyncSubnets(
...
-		}
-	}
-
 	return true
 }
 
-// subscribeDynamicWithSyncSubnets subscribes to a dynamically changing list of subnets.
-func (s *Service) subscribeDynamicWithSyncSubnets(
+// subscribeSubnets subscribes to a list of subnets.
+func (s *Service) subscribeSubnets(
```
Comment (nisdas) [inline beacon-chain/sync/subscriber.go:460]: can you rename this or `subscribeToSubnets` ? They sound too similar to each other which makes it easy to confuse between both of them

**Review by nisdas — APPROVED** [review]

**Pull request merged**